src/picongpu/include/particles/ionization/byField/BSI/BSI.hpp

```diff
@@ -32,30 +45,147 @@ namespace particles
 namespace ionization
 {
 
-    /** \struct BSI
-     * \brief Barrier Suppression Ionization
+    /** \struct BSI_Impl
+     * \brief Barrier Suppression Ionization - Implementation
+     *
+     * \tparam T_DestSpecies electron species to be created
+     * \tparam T_SrcSpecies particle species that is ionized
      */
-    template<typename T_DestSpecies>
-    struct BSI
+    template<typename T_DestSpecies, typename T_SrcSpecies>
+    struct BSI_Impl
```

Comment:
Please rename this file. Files should have the same name like the internal class. In this case `BSI_Impl.hpp`